.github/workflows/first_contrib_cert_generator.yml

```diff
@@ -0,0 +1,270 @@
+    runs-on: ubuntu-latest
+    steps:
+      # Step 1: Check if this is the contributor's first merged PR.
+      # This step is the source of truth and will control the execution of subsequent steps.
+      - name: Check for first merged PR
+        id: check_first_pr
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const author = context.payload.pull_request.user.login;
+            const query = `repo:${context.repo.owner}/${context.repo.repo} is:pr is:merged author:${author}`;
+            
+            console.log(`Searching for merged PRs from @${author} with query: "${query}"`);
+            
+            const result = await github.rest.search.issuesAndPullRequests({ q: query });
+            const mergedPRs = result.data.total_count;
+
+            if (mergedPRs === 1) {
+              console.log(`SUCCESS: This is the first merged PR from @${author}. Proceeding...`);
+              core.setOutput('is_first_pr', 'true');
+            } else {
+              console.log(`INFO: Skipping certificate generation. @${author} has ${mergedPRs} total merged PRs.`);
+              core.setOutput('is_first_pr', 'false');
+            }
+      
+      # Step 2: Checkout the repository containing the certificate HTML file.
+      - name: Checkout containers/automation repository
+        if: ${{ github.event_name == 'workflow_dispatch' || steps.check_first_pr.outputs.is_first_pr == 'true' }}
+        uses: actions/checkout@v4
+        with:
+          repository: containers/automation
+          path: automation-repo
+
+      # Step 3: Update the HTML file locally
+      - name: Update HTML file
+        if: ${{ github.event_name == 'workflow_dispatch' || steps.check_first_pr.outputs.is_first_pr == 'true' }}
+        run: |
+          HTML_FILE="automation-repo/certificate-generator/certificate_generator.html"
+          CONTRIBUTOR_NAME="${{ github.event.inputs.contributor_username || github.event.pull_request.user.login }}"
+          PR_NUMBER="${{ github.event.inputs.pr_number || github.event.pull_request.number }}"
+          MERGE_DATE=$(date -u +"%B %d, %Y")
+
+          sed -i "/id=\"contributorName\"/s/value=\"[^\"]*\"/value=\"${CONTRIBUTOR_NAME}\"/" ${HTML_FILE} || { echo "ERROR: Failed to update contributor name."; exit 1; }
+          sed -i "/id=\"prNumber\"/s/value=\"[^\"]*\"/value=\"#${PR_NUMBER}\"/" ${HTML_FILE} || { echo "ERROR: Failed to update PR number."; exit 1; }
+          sed -i "/id=\"mergeDate\"/s/value=\"[^\"]*\"/value=\"${MERGE_DATE}\"/" ${HTML_FILE} || { echo "ERROR: Failed to update merge date."; exit 1; }
+
+      # Step 4: Setup Node.js environment
+      - name: Setup Node.js
+        if: ${{ github.event_name == 'workflow_dispatch' || steps.check_first_pr.outputs.is_first_pr == 'true' }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
```

Comment:
Why is this using an outdated version? This is already EOL https://nodejs.org/en/eol